k8s/cmd/commands/util/parse.go

```diff
@@ -16,16 +16,72 @@ limitations under the License.
 package util
 
 import (
+	"fmt"
+	"io"
+	"os"
 	"strings"
 
 	"github.com/goccy/go-json"
+	"sigs.k8s.io/yaml"
 
 	"github.com/pkg/errors"
 
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
+// ConvertToJson check whether the string is json type and
+// converts the string to json type if not
+func ConvertToJson(str string) (string, error) {
+	result := make(map[string]interface{})
+	err := json.Unmarshal([]byte(str), &result)
+	fmt.Println("str is ", str)
+	if err != nil {
+		json, err := yaml.YAMLToJSON([]byte(str))
+		if err != nil {
+			return "", fmt.Errorf("failed to convert to json: %v", err)
+		}
+		return string(json), nil
+	}
+	return str, nil
+}
+
+// ConvertToYaml converts the json string to yaml type
+func ConvertToYaml(str string) (string, error) {
+	// Unmarshal the JSON string into a map
+	var m map[string]interface{}
+	err := json.Unmarshal([]byte(str), &m)
+	if err != nil {
+		return "", fmt.Errorf("failed to unmarshal the json: %v", err)
+	}
+
+	// Marshal the map into a YAML string
+	y, err := yaml.Marshal(m)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal to yaml: %v", err)
+	}
+
+	return string(y), nil
+}
+
+// ParseFromStdin parse the stdin to string
+func ParseFromStdin(args []string) (string, error) {
+	// Check if the input is coming from '-'
+	if len(args) > 0 && args[0] == "-" {
+		// Read all the bytes piped in from stdin
+		r, err := io.ReadAll(os.Stdin)
+		if err != nil {
+			return "", fmt.Errorf("failed to read from stdin: %v", err)
+		}
+		j, err := ConvertToJson(string(r))
+		if err != nil {
+			return "", fmt.Errorf("failed to convert to json: %v", err)
+		}
+		return j, nil
+	}
+	return "", nil
+}
```

Comment:
Don't mix unrelated things. You could have a function that only do the read. Otherwise you will need to copy this part to handle reading YAMLs from stdin someday later.